src/NServiceBus.Core/Persistence/InMemory/Outbox/InMemoryOutboxStorage.cs

```diff
@@ -31,7 +30,7 @@ public Task Store(OutboxMessage message, OutboxTransaction transaction, ContextB
             var tx = (InMemoryOutboxTransaction) transaction;
             tx.Enlist(() =>
             {
-                if (!storage.TryAdd(message.MessageId, new StoredMessage(message.MessageId, message.TransportOperations.ToList())))
```

Comment:
Should be able to remove this `ToArray()` call since `TransportOperations` is already an array now, or is there some reason you need a copy of the collection?